### DISCUSSION TITLE
Syntax highlighting for shell commands

### DISCUSSION THREAD

**Discussion by Taitava** [Taitava/obsidian-shellcommands#101] [Ideas]
There could be syntax highlighting for shell commands, meaning that words would be classified and colored based their classes. This is a really basic feature in programming editors.

Places where this could be used:
 - Shell command input fields *
 - Preview texts below the input fields *
 - Command palette *
 - Execution confirmation modal
 - Deletion confirmation modal

*) These might require hard research to find out how their visual content structure could be accessed and modified.

It's not needed to leave the Obsidian ecosystem to find an example for this feature. [Deathau's Editor syntax highlight Obsidian plugin](https://github.com/deathau/cm-editor-syntax-highlight-obsidian) does a similar thing for code blocks, and I used it to create a small example of how syntax highlighting could look with shell commands:

![kuva](https://user-images.githubusercontent.com/13002142/140696389-0ee68924-4a89-412c-a08b-57639652c658.png)

Of course the difference with SC would be, that syntax highlighting would be single line only, but the general idea can be seen in the image.

_Editor syntax highlight_ uses [CodeMirror's syntax highlighting modes](https://github.com/codemirror/CodeMirror/tree/5.58.2/mode). I think SC could adapt a subset of those, too - at least [shell](https://github.com/codemirror/CodeMirror/tree/5.58.2/mode/shell) and [powershell](https://github.com/codemirror/CodeMirror/tree/5.58.2/mode/powershell).

**Comment by FelipeRearden** [comment]
That's a great idea @Taitava !
